src/daft-physical-plan/src/physical_planner/planner.rs

```diff
@@ -496,6 +402,7 @@ pub struct AdaptivePlanner {
     remaining_logical_plan: Option<LogicalPlanRef>,
     remaining_physical_plan: Option<PhysicalPlanRef>,
     last_stage_id: usize,
+    last_stage_clustering_spec: Option<ClusteringSpecRef>,
```

Comment:
Is there a reason we moved out the clustering spec out of the PlaceHolder to here?